exercise/do-your-own-map.js

```diff
@@ -0,0 +1,113 @@
+/*
+## Do-Your-Own-Map
+
+### Instructions
+
+1) Create a `map` function that takes an array as first argument, a function as second,
+and that works like the method .map
+
+2) Create a `flatMap` function that takes an array as first argument, a function as second,
+and that works like the method .flatMap
+
+### Notions
+
+- Map
+https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map
+
+- FlatMap
+
+*/
+Array.prototype.map = undefined
+Array.prototype.flatMap = undefined
+
+///*/ // ⚡
+
+///*/// ⚡
+
+export const tests = []
+const t = (f) => tests.push(f)
+
+let arr1 = [10, -10, 20, -95, 86, 102, 35, 89, 110]
+let arr2 = [[10], -10, 20, -95, 86, [102], [35, 89], 110]
+let arr3 = [[5], [4], [-3], [20], [17], [-33], [-4], [18]]
+
+Object.freeze(arr1)
+Object.freeze(arr2)
+Object.freeze(arr3)
+
+const add1 = (el) => el + 1
+const mult2 = (el) => el * 2
+const sub3 = (el) => el - 3
+
+const doAll = (el) => sub3(mult2(add1(el)))
+const posValsIndex = (el, i) => (el >= 0 ? `${i}: ${el}` : null)
```

Comment:
please refrain for using `null` as much as possible, use `undefined`.

Reply:
read more about why: https://github.com/sindresorhus/meta/issues/7 (more in the comments)

Reply:
ok, :) I will refrain from using `null` in my JS code. Interesting thread, did not expect to see so many opinions on the subject  :O  
All `null` have been replaced by `undefined`